src/net/NetworkClient.cpp

```diff
@@ -0,0 +1,54 @@
+#include "NetworkClient.h"
+
+NetworkClient::NetworkClient(boost::asio::io_service* ios) :
+    io_service(ios), socket(*ios), connected(false) {
+}
+
+bool NetworkClient::connect(std::string& host, unsigned int port) {
+    tcp::resolver resolver(*io_service);
+    tcp::resolver::query query(tcp::v4(), host, std::to_string(port));
+    tcp::resolver::iterator endpoint_iterator = resolver.resolve(query);
+    try {
+        boost::asio::connect(socket, endpoint_iterator);
+    } catch (...) {
+        return false;
+    }
+
+    connected = true;
+    start_receive();
+    return true;
+}
+
+bool NetworkClient::is_connected() const {
+    return connected;
+}
+
+bool NetworkClient::send(std::string& message) {
+    if (!connected)
+        return false;
+    boost::asio::write(socket, boost::asio::buffer(message));
+    return true;
+}
+
+bool NetworkClient::read_message(std::string& message) {
```

Comment:
This is actually an instance where it is better to not use a reference - by making it a pointer, it's more clear that it is an input parameter.